addons/project/models/res_partner.py

```diff
@@ -61,3 +61,34 @@ def _create_portal_users(self):
                 'active': True,
             })
         return created_users
+
+    def action_view_tasks(self):
+        self.ensure_one()
+        action = self.env["ir.actions.actions"]._for_xml_id("project.action_view_all_task")
+        action['display_name'] = _("%(partner_name)s's Tasks", partner_name=self.name)
+        action['context'] = {
+            'default_partner_id': self.id,
+        }
+        views = []
+        for view in action['views']:
+            if view[1] == 'form':
+                views.append((self.env.ref('project.view_task_form_res_partner').id, view[1]))
+            elif view[1] == 'kanban':
+                views.append((self.env.ref('project.view_task_kanban_res_partner').id, view[1]))
+            else:
+                views.append(view)
+        action['views'] = views
+        all_child = self.with_context(active_test=False).search([('id', 'child_of', self.ids)])
```

Comment:
```suggestion
        all_child = self.with_context(active_test=False).child_ids
```
Doesn't that work?

Reply:
As we also need to get the inactive childs (i.e. archived res_partner), I think we need to keep the search(). I tried without it and it didn't include the inactive records :/